types/imagekit/index.d.ts

```diff
@@ -0,0 +1,280 @@
+// Type definitions for imagekit 3.1
+// Project: https://github.com/imagekit-developer/imagekit-nodejs
+// Definitions by: Kevin Faulhaber <https://github.com/kemicofa>
+//                 Romanos Tsouroplis <https://github.com/romdim>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="node" />
+
+type TransformationPosition = 'path' | 'query';
+type FileType = 'all' | 'image' | 'non-image';
+type Item = 'file' | 'folder';
+type CustomCoordinatesResponse = string | null;
+// Taken from this help post: https://help.imagekit.io/en/articles/2434102-image-format-support-in-imagekit-for-resizing-compression-and-static-file-delivery
+type FileFormat = 'jpg' | 'png' | 'gif' | 'svg' | 'webp' | 'pdf' | 'js' | 'css' | 'txt' | 'mp4' | 'webm' | 'mov' | 'swf' | 'ts' | 'm3u8' | string;
+
+interface Transformation {
+    height?: string;
+    width?: string;
+    aspectRatio?: string;
+    quality?: string;
+    crop?: string;
+    cropMode?: string;
+    x?: string;
+    y?: string;
+    focus?: string;
+    format?: string;
+    radius?: string;
+    background?: string;
+    border?: string;
+    rotation?: number;
+    blur?: string;
+    named?: string;
+    overlayImage?: string;
+    overlayX?: string;
+    overlayY?: string;
+    overlayFocus?: string;
+    overlayHeight?: string;
+    overlayWidth?: string;
+    overlayText?: string;
+    overlayTextFontSize?: string;
+    overlayTextColor?: string;
+    overlayAlpha?: string;
+    overlayTextTypography?: string;
+    overlayBackground?: string;
+    overlayImageTrim?: string;
+    progressive?: string;
+    lossless?: string;
+    trim?: string;
+    metadata?: string;
+    colorProfile?: string;
+    defaultImage?: string;
+    dpr?: string;
+    effectSharpen?: string;
+    effectUSM?: string;
+    effectContrast?: string;
+    effectGray?: string;
+    original?: string;
+}
+
+interface UrlOptionsBase {
+    transformation?: Transformation[];
+    transformationPosition?: TransformationPosition;
+    queryParameters?: { [key: string]: string | number };
+    urlEndpoint?: string;
+    signed?: boolean;
+    expireSeconds?: number;
+}
+
+type UrlOptions = ({ src: string; path?: never } | { path: string; src?: never }) & UrlOptionsBase;
+
+interface UploadOptions {
+    file: number | string;
+    fileName: string;
+    useUniqueFileName?: boolean;
+    tags?: string;
+    folder?: string;
+    isPrivateFile?: boolean;
+    customCoordinates?: string;
```

Comment:
That's why I introduced the customCoordinatesResponse type, as only the response can be null.
I've tested it and it breaks when I pass `customCoordinates: null` within the upload function.

And yes, maybe we should do the same for tags as it has same behavior.
I know it's not ideal but it's how their api works unfortunately.

Reply:
For the ListFileResponse indeed, I saw it through testing even though they don't have it in their docs and it's my To-do's to tell them about it.

For the customCoordinates, what about NullableCustomCoordinates then?
Otherwise I can just remove the type altogether and use the null wherever needed and that's that.
Either is fine by me!